src/main/java/org/terasology/dynamicCities/settlements/SettlementEntityManager.java

```diff
@@ -155,9 +155,9 @@ public void update(float delta) {
             settlementEntities = settlementCachingSystem.getSettlementCacheEntity();
         }
 
-        counter--;
-        timer++;
-        if (counter != 0) {
+        counter -= delta;
```

Comment:
The word `counter` does not really match what it's doing anymore, so I would rename it. I also think that it should change according to world time. You don't want it to change while the game is paused, do you?

Reply:
Should it use `WorldTimeEvent`? That event fires every 0.01 unit of world time, that would be nice for the city cycle

Reply:
I changed it to use world time instead 👍